### PR TITLE
Revert Table and refactor implementation, fix drag issues

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
Follow up of https://github.com/opendatahub-io/odh-dashboard/pull/1177, and part of the fix covers #1185 

## Description

Revert the Table component, use the draggable hook in the custom serving runtime list view component instead.

## How Has This Been Tested?

Mostly check the code, drag and drop the serving runtimes table, add and delete them and make sure nothing is broken. This PR adds no new features, mainly code refactoring and small bug fixes.

## Test Impact

N/A

## Request review criteria:

- [x] The commits have meaningful messages (squashes happen on merge by the bot).
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

### CODE REVIEW

File: frontend/src/pages/modelServing/customServingRuntimes/CustomServingRuntimeListView.tsx [hunk 1]
```diff
@@ -1,59 +1,65 @@
...
-  React.useEffect(() => {
-    patchDashboardConfigTemplateOrder(dragItemOrder, dashboardNamespace).catch((e) =>
-      notification.error(`Error update the serving runtimes order`, e.message),
-    );
-  }, [dragItemOrder, dashboardNamespace, notification]);
-
-  React.useEffect(() => {
-    setDragItemOrder(getDragItemOrder(templates, templateOrder));
-  }, [templates, templateOrder]);
+  const { isDragging, tbodyDragFunctions, trDragFunctions } = useDraggableTable(
```
Comment (andrewballantyne) [inline frontend/src/pages/modelServing/customServingRuntimes/CustomServingRuntimeListView.tsx:41]: Recommend that you look to see how you can put everything inside the `useDraggableTable`... your hook should set you up for _everything_ you need. The output of a hook could be a series of `row` and `table` props -- any use of those props could be entirely managed within' the hook itself.

Highly recommend you take a look at your component here and understand the setup you're doing here in the `CustomServingRuntimeListView` for this hook -- can not all of this be within it?

**Review by lucferbux — APPROVED** [review]
/lgtm

This approach adds LSP to the table component (extending the props of the table and the row)
@andrewballantyne I'm not sure if this is what you had in mind, but the logic seems to be getting both props and passing them to the elements.
Rest of the logic seems fine to me.

**Comment by andrewballantyne** [comment]
I'm not running model serving today -- but the interface to the table looks good to me. Sub props is uncommon, but not that problematic... But now that I think of it... `disableRowRenderSupport` and the props are in direct conflict. We may want to use `EitherNotBoth` generic type to help prevent passing both.

I assume it works and covers the cases.

**Comment by lucferbux** [comment]
> I'm not running model serving today -- but the interface to the table looks good to me. Sub props is uncommon, but not that problematic here (since 'tbody' is a known semantic part of 'table')... But now that I think of it... `disableRowRenderSupport` and the props are in direct conflict. We may want to use `EitherNotBoth` generic type to help prevent passing both.
> 
> I assume it works and covers the cases.

Ok, perfect, thanks for the feedback, @DaoDaoNoCode we can review it tomorrow

**Review by lucferbux — APPROVED** [review]
/lgtm

File: frontend/src/components/Table.tsx [hunk 2]
```diff
@@ -22,9 +23,11 @@ type TableProps<DataType> = {
   toolbarContent?: React.ReactElement<typeof ToolbarItem>;
   emptyTableView?: React.ReactElement<typeof Tr>;
   caption?: string;
-  disableRowRenderSupport?: boolean;
-  tbodyProps?: TbodyProps & { ref?: React.Ref<HTMLTableSectionElement> };
-} & Omit<TableComposableProps, 'ref' | 'data'>;
+} & EitherNotBoth<
+  { disableRowRenderSupport?: boolean },
+  { tbodyProps?: TbodyProps & { ref?: React.Ref<HTMLTableSectionElement> } }
```
Comment (andrewballantyne) [inline frontend/src/components/Table.tsx:28]: If you wanted to -- you could make this EitherOrNone -- and make them required. Just realized that by looking at it now... sorry about that.

**Pull request merged**